fake-dependency-service/src/main/kotlin/co/tala/api/fakedependency/business/helper/RequestExtractor.kt

```diff
@@ -1,15 +1,25 @@
 package co.tala.api.fakedependency.business.helper
 
 import co.tala.api.fakedependency.configuration.helper.RequestIdExtractorConfiguration
+import co.tala.api.fakedependency.redis.IRedisService
+import co.tala.api.fakedependency.redis.RedisKeyPrefix
+import com.fasterxml.jackson.core.type.TypeReference
+import com.fasterxml.jackson.databind.ObjectMapper
 import org.springframework.stereotype.Component
 import org.springframework.util.LinkedMultiValueMap
 import javax.servlet.http.HttpServletRequest
 
 @Component
 class RequestExtractor(
     private val keyHelper: IKeyHelper,
-    private val config: RequestIdExtractorConfiguration
+    private val config: RequestIdExtractorConfiguration,
+    private val redisSvc: IRedisService,
+    private val objectMapper: ObjectMapper
 ) : IRequestExtractor {
+    companion object {
+        private const val X_FAKE_DEPENDENCY_PARSE_PAYLOAD_HEADER: String = "X-Fake-Dependency-Parse-Payload-Header"
+    }
```

Comment:
Not for this change, but curious, would we want this to handle any custom header? So we could define environment variable, which could be a list of headers to parse? Not sure there is a use case yet, outside of SageMaker...

Reply:
@autumn-kennedy yes, this is a custom header, but in this case it's not defined by application property. It's defined in the api call itself. Please see the [ParseRequestHeaderPayload.http](http/ParseRequestHeaderPayload.http) for examples on how to define the custom named header to parse.